pkg/schema/load/load_test.go

```diff
@@ -0,0 +1,96 @@
+package load
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/grafana"
+	"github.com/grafana/grafana/pkg/schema"
+	"github.com/stretchr/testify/require"
+)
+
+var p BaseLoadPaths = BaseLoadPaths{
+	BaseCueFS:       grafana.CoreSchema,
+	DistPluginCueFS: grafana.PluginSchema,
+}
+
+// Basic well-formedness tests on core scuemata.
+func TestScuemataBasics(t *testing.T) {
+	all := make(map[string]schema.VersionedCueSchema)
+
+	dash, err := BaseDashboardScuemata(p)
+	require.NoError(t, err, "error while loading base dashboard scuemata")
+	all["basedash"] = dash
+
+	ddash, err := DistDashboardScuemata(p)
+	require.NoError(t, err, "error while loading dist dashboard scuemata")
+	all["distdash"] = ddash
+
+	for set, sch := range all {
+		t.Run(set, func(t *testing.T) {
+			if sch == nil {
+				t.Error("scuemata linked to empty chain")
+			}
+
+			maj, min := sch.Version()
+			t.Run(fmt.Sprintf("%v.%v", maj, min), func(t *testing.T) {
+				cv := sch.CUE()
+				t.Run("Exists", func(t *testing.T) {
+					require.True(t, cv.Exists(), "cue value for schema does not exist")
+				})
```

Comment:
Subtests (t.Run) should only be used for distinct test cases, not to group assertions.
```suggestion
				require.True(t, cv.Exists(), "cue value for schema does not exist")
```

Reply:
Why's that?

i've argued elsewhere that subtests can and even should be used to generate test matrixes in a style reminiscent of [property testing](https://en.wikipedia.org/wiki/Property_testing).

Now, i can see an argument for why that doesn't apply here - if `Exists()` fails, then `Validate()` will also necessarily fail, so separating them into two distinct subtests doesn't tell us new things.  But CUE is young and has warts, and i don't necessarily trust that will always be the case - and since this particular test is intended to check really basic behavioral properties of any scuemata family, it's not a bad idea to check both independently. (Maybe there should even be a third, independent subtest that ensures we don't have validation passing when existence fails.)